lib/sprockets/digest_utils.rb

```diff
@@ -84,6 +84,39 @@ def detect_digest_class(bytes)
     }
     private_constant :ADD_VALUE_TO_DIGEST
 
+    def a_v_t_d(c, v, d)
```

Comment:
this isn't significantly faster, only 10% on ruby 2.4 and even slower on older versions

Reply:
oh, interesting, it makes the digest completely disappear from my callgraph, saving 30%,
what os, cpu are did you try on?

Reply:
well, if 10% is not significant ...

Reply:
10% improvement looks nice, but if this call takes for instance only 5% of the entire compilation process, than it's only a microoptimization without any significant effect.

I was wondering how fast this digest optimization is, so I tested it on our pipeline (big project) and here're my numbers (spent time in this method, same hw)
**ruby 2.4.2 windows 29,4% faster**
**ruby 2.4.2 linux 17,1% faster**
ruby 2.1.9 windows 8,6% slower
**ruby 2.1.9 linux 17.6% faster**
jruby 9.1.13.0 windows 10,3% slower
jruby 9.1.13.0 linux 11,1% slower

I didn't have time to dig deeper. In my opinion your implementation should be faster, but optimizations are sometimes tricky :)

Reply:
Thanks for the numbers, terrific :)
With the compilation time going down 10 times, the digest became very well visible in the call graph, therefore i looked at it, because that small improvement became more significant with overall times going down that much.

ruby 2.1.* is obsolete since spring 2017

jruby only supports rails 4, which is, except for severe security issues, out of support too
And if a case is slower on jruby, and if this project really wants to support jruby, this may be a good case for opening a 'performance: case' ticket over at jruby. :)

Reply:
Originally I optimized this using refinements, it's even faster than what we ended up with in MRI. But it made it way slower on jruby.

If you're going to do microbenchmarks, please use benchmark-ips as in

-    #312
-   #383
-   #417

Last time we looked at this the hash method was faster than the case method. A big improvement was added when we started using compare with identity https://github.com/rails/sprockets/pull/439/files.